homeassistant/components/unifiprotect/media_source.py

```diff
@@ -0,0 +1,831 @@
+
+        camera: Camera | None = None
+        if camera_id != "all":
+            camera = data.api.bootstrap.cameras.get(camera_id)
+            args["camera_id"] = camera_id
+
+        title = f"{start.strftime('%B %Y')} > {title}"
+        events = await self._build_events(**args)  # type: ignore[arg-type]
+        source.children = events  # type: ignore[assignment]
+        source.title = self._breadcrumb(
+            data,
+            title,
+            camera=camera,
+            event_type=event_type,
+            count=len(events),
+        )
+
+        return source
+
+    async def _build_events_type(
+        self,
+        data: ProtectData,
+        camera_id: str,
+        event_type: SimpleEventType,
+        build_children: bool = False,
+    ) -> BrowseMediaSource:
+        """Build folder media source for a selectors for a given event type."""
+
+        base_id = f"{data.api.bootstrap.nvr.id}:{camera_id}:{event_type.name.lower()}"
+
+        title = event_type.value.title()
+        source = BrowseMediaSource(
+            domain=DOMAIN,
+            identifier=base_id,
+            media_class=MEDIA_CLASS_DIRECTORY,
+            media_content_type=VIDEO_FORMAT,
+            title=title,
+            can_play=False,
+            can_expand=True,
+            children_media_class=MEDIA_CLASS_VIDEO,
+        )
+
+        if not build_children or data.api.bootstrap.recording_start is None:
+            return source
+
+        children = [
+            self._build_recent(data, camera_id, event_type, 1),
+            self._build_recent(data, camera_id, event_type, 7),
+            self._build_recent(data, camera_id, event_type, 30),
+        ]
+
+        start, end = _get_start_end(self.hass, data.api.bootstrap.recording_start)
+        while end > start:
+            children.append(self._build_month(data, camera_id, event_type, end.date()))
+            end = (end - timedelta(days=1)).replace(day=1)
+
+        camera: Camera | None = None
+        if camera_id != "all":
+            camera = data.api.bootstrap.cameras.get(camera_id)
+        source.children = await asyncio.gather(*children)
+        source.title = self._breadcrumb(data, title, camera=camera)
+
+        return source
+
+    async def _get_camera_thumbnail_url(self, camera: Camera) -> str | None:
+        """Get camera thumbnail URL using the first available camera entity."""
+
+        if not camera.is_connected or camera.is_privacy_on:
+            return None
+
+        entity_id: str | None = None
+        entity_registry = await self.get_registry()
+        for channel in camera.channels:
+            base_id = f"{camera.mac}_{channel.id}"
+            entity_id = entity_registry.async_get_entity_id(
+                Platform.CAMERA, DOMAIN, base_id
+            )
+            if entity_id is None:
+                entity_id = entity_registry.async_get_entity_id(
+                    Platform.CAMERA, DOMAIN, f"{base_id}_insecure"
+                )
```

Comment:
Since I know this will come up. This is essentially a greed algorithm. Since we cannot know which camera channels the user has enabled, we cannot just generate a valid unique ID. That being said, this for loop can run `async_get_entity_id` 1 to 6 times, but _should_ run only 1 _most_ of the time if the user has the default camera entity enabled (High RTSPS). 

Worse case is if the user does not have Low RTSP only enabled at all on a G4 Doorbell Pro, it will call `async_get_entity_id` 6 times for the camera.